### ISSUE TITLE
Allow specifying a command line instead of just a path for plugins in `buf.gen.yaml`

### ISSUE THREAD

**Issue by espadolini** [bufbuild/buf#1557]
It would be neat if the `path` of a plugin could also include some arguments, so that it would be possible to do
```yaml
plugins:
  - name: go
    path: ["go", "run", "google.golang.org/protobuf/cmd/protoc-gen-go"]
    out: gen/proto
    opt: paths=source_relative
```

to always run the version of `protoc-gen-go` that matches the `google.golang.org/protobuf` in use (as specified by `go.mod`).

**Comment by advanderveer** [comment]
I came here for the same functionality. Not sure about the implementation but the reasoning for this is:
- Developers might use different versions of plugins between projects, so a single version in the PATH is inflexible
- This is super useful for plugin development (not having to recompile manually into the PATH every time)
- It is recommended to version tooling of a Go project in a tools.go file with a build constraint, the possibility to use go run with this makes it so that plugin versioning happens transparently (and without surprises)

**Comment by bufdev** [comment]
This seems like a reasonable request. For future implementers, what you need to do:

- Update `ExternalPluginConfigV1` in `bufgen.go` to take either a `string` or `[]string` (there's a bunch of ways to do this, you'll have to implement `UnmarshalJSON` on a custom type likely, or I think you can use `json.RawMessage`)
- Some validation in `bufgen` will need to change.
- The `GenerateWithPluginPath` option in `appprotoexec` will need to take `...string` and the documentation will need to be updated for this option.
- Same for `HandlerWithPluginPath`.
- `binaryHandler` will need to call `RunWithArgs` option if the path is of length > 1.
- Testing.
- Documentation.

Run this one by someone who deeply understands the generation stack if this is implemented.

**Comment by bufdev** [comment]
This is released into v1.14.0.
